storage/blockdevice/tests/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -801,6 +804,9 @@ void test_unaligned_erase_blocks()
     // Valid erase
     err = block_device->erase(addr, sector_erase_size);
     TEST_ASSERT_EQUAL(0, err);
+
+    err = block_device->deinit();
+    TEST_ASSERT_EQUAL(0, err);
```

Comment:
This will cause the same problem (uninitialised block device) in [test_rww_rwe](https://github.com/ARMmbed/mbed-os/blob/41675e16fa357e90b7a4acd093a862c4899b0f05/storage/blockdevice/tests/TESTS/blockdevice/general_block_device/main.cpp#L909) if MX_FLASH_SUPPORT_RWW.

Reply:
Shouldn't each test init/deinit the block device? This would be more consistent with how the API should be used.

Reply:
> Shouldn't each test init/deinit the block device? This would be more consistent with how the API should be used.

This would be the most ideal in my opinion. But test cases in this file are dependent on one another, for example test_deinit_bd depends on previous tests that initialised the block device. If we want to make each test case standalone, a bit of rework would be needed. Otherwise the minimal change is to keep the current logic.